hashes/src/sha256t.rs

```diff
@@ -112,6 +112,7 @@ fn from_engine<T: Tag>(e: sha256::HashEngine) -> Hash<T> {
 ///
 /// [`hash_newtype`]: crate::hash_newtype
 #[macro_export]
+#[deprecated(since = "TBD", note = "this macro is no longer supported, will be removed soon")]
```

Comment:
In 57a9325fceb261c700db0888aa3724fa42af262e:

What if, instead of deprecating these, we made them a wrapper around the more detailed macros, where the more-detailed ones would be called by pseudo-attributes like `#[hash_impl(hex_fmt)]` etc. And have the `hash_wrapper` macro always be called.

This would minimize the amount of work we'd have to do to make the "hash invocation looks like real code" syntax, and minimize the verbosity of calling these things. Though it would be a bit more work in *this* macro to support arbitrary sets of attributes like this.